File: src/main/java/eotree/data/EOIntData.java
package eotree.data;

/**
 * EBNF representation:
 * <code>
 * /[+-]?\d+|0x[a-f\d]+/
 * </code>
 */
public class EOIntData extends EOData {
    public Integer i;

    public EOIntData(Integer i) {
        this.i = i;
    }

    @Override
    public String generateEO(int indent) {
        return indent(indent) + i.toString();
    }
}

File: src/main/java/eotree/data/EOIntData.kt
package eotree.data

/**
 * EBNF representation:
 * `
 * /[+-]?\d+|0x[a-f\d]+/
` *
 */
class EOIntData(var i: Int) : EOData() {
    override fun generateEO(indent: Int): String {
        return indent(indent) + i.toString()
    }
}
